modules/catalog-compaction/src/main/java/org/apache/ignite/internal/catalog/compaction/CatalogManagerCompactionFacade.java

```diff
@@ -84,6 +86,22 @@ Int2IntMap collectTablesWithPartitionsBetween(long minTsInclusive, long maxTsInc
         }
     }
 
+    /**
+     * Returns the catalog version that is active at the given timestamp.
+     *
+     * @param timestamp Timestamp.
+     * @return Catalog revision or {@code null} if such version of the catalog doesn't exist.
+     */
+    @Nullable Catalog catalogAtTsNullable(long timestamp) {
```

Comment:
This method should be called `catalogAtTs()` it should never return null, because its invocations always follow not null checks.